packages/run-protocol/src/feeDistributor.js

```diff
@@ -0,0 +1,302 @@
+// @ts-check
+
+import { AmountMath } from '@agoric/ertp';
+import { E, Far } from '@endo/far';
+import { observeNotifier } from '@agoric/notifier';
+import { makeScalarMapStore } from '@agoric/store';
+
+/**
+ * @typedef {object} FeeCollector
+ *
+ * @property {() => ERef<Payment>} collectFees
+ */
+
+/**
+ * @typedef {object} DistributorParams
+ *
+ * @property {bigint} [epochInterval=1n] - parameter to the epochTimer
+ *  controlling the interval at which rewards should be sent to the bank.
+ */
+
+/**
+ * @callback BuildFeeDistributor
+ *
+ * @param {ERef<FeeCollector>[]} collectors - an array of objects with
+ *   collectFees() methods, each of which will return a payment. Can
+ *   be populated with the results of makeFeeCollector(zoe, creatorFacet)
+ * @param {EOnly<DepositFacet>} feeDepositFacet - object with receive()
+ * @param {ERef<TimerService>} epochTimer - timer that notifies at the end of
+ *  each Epoch. The epochInterval parameter controls the interval.
+ * @param {DistributorParams} params
+ * @returns {Promise<void>}
+ */
+
+/**
+ * @typedef {object} CollectibleContractFacet
+ * @property {() => Promise<Invitation>} makeCollectFeesInvitation
+ */
+
+/**
+ * wrapper to take the vaultFactory or AMM's creatorFacet, and make a function that
+ * will request an invitation and return a promise for a payment.
+ *
+ * @param {ERef<ZoeService>} zoe
+ * @param {ERef<CollectibleContractFacet>} creatorFacet
+ */
+export function makeFeeCollector(zoe, creatorFacet) {
+  /** @type {FeeCollector} */
+  return Far('feeCollector', {
+    collectFees: () => {
+      const invitation = E(creatorFacet).makeCollectFeesInvitation();
+      const collectFeesSeat = E(zoe).offer(invitation, undefined, undefined);
+      return E(collectFeesSeat).getPayout('RUN');
+    },
+  });
+}
+
+/**
+ * A distributor of fees from Inter Protocol sources to deposit facets. Each
+ * time the epochTimer signals the end of an Epoch, it will ask the contracts
+ * for fees that have been collected to date and send that payment to the
+ * depositFacet.
+ *
+ * @param {() => Promise<any>} schedulePayments - distribute to the destinations
```

Comment:
fully typing this isn't worth it, but `unknown` is always safer than `any`.
```suggestion
 * @param {() => Promise<unknown>} schedulePayments - distribute to the destinations
```